src/base59.cpp

```diff
@@ -0,0 +1,146 @@
+#include <assert.h>
+#include <limits>
+#include <stdexcept>
+#include <string.h>
+#include <vector>
+#include <veriblock/base58.hpp>
+#include <string>
+
+namespace VeriBlock {
+
+static std::string g_Base59Alphabet =
+    "123456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz0";
+static const uint32_t g_kBase_256 = 256;
+static const size_t g_kBase59 = g_Base59Alphabet.size();
+
+static const std::vector<int8_t> g_Indexes = {
+-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+-1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1, -1,
+-1, -1, -1, -1, -1, -1, -1, -1, 58, 0, 1, 2, 3, 4, 5, 6, 7, 8, -1, -1, -1, -1, -
+1, -1, -1, 9, 10, 11, 12, 13, 14, 15, 16, -1, 17, 18, 19, 20, 21, -1, 22, 23, 24
+, 25, 26, 27, 28, 29, 30, 31, 32, -1, -1, -1, -1, -1, -1, 33, 34, 35, 36, 37, 38
+, 39, 40, 41, 42, 43, -1, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57
+, -1, -1, -1, -1, -1 };
+
+
+static uint8_t divmod59(std::vector<uint8_t> &number, size_t startAt) {
+    size_t remainder = 0;
+  for (size_t i = startAt; i < number.size(); i++) {
+    uint32_t digit256 = (uint32_t)number[i] & 0xFF;
+    size_t temp = remainder * g_kBase_256 + digit256;
+    number[i] = (uint8_t)(temp / g_kBase59);
+    remainder = temp % g_kBase59;
+  }
+  return (uint8_t)remainder;
+}
+
+static uint8_t divmod256(std::vector<uint8_t> &number59, size_t startAt) {
+    size_t remainder = 0;
+  for (size_t i = startAt; i < number59.size(); i++) {
+    int digit59 = (int8_t)number59[i] & 0xFF;
+    size_t temp = remainder * g_kBase59 + digit59;
+    number59[i] = (uint8_t)(temp / g_kBase_256);
+    remainder = temp % g_kBase_256;
+  }
+  return (uint8_t)remainder;
+}
+
+std::string EncodeBase59(const uint8_t *buf, size_t nSize) {
+
+  if (!nSize) {
+    return std::string();
+  }
+
+  // Make a copy of the input since we are going to modify it.
+  std::vector<uint8_t> input(buf, buf+nSize);
+  if (input.empty()) {
+    return std::string();
+  }
```

Comment:
this check is redundant, because you check that size is 0, and input can be 0 only if `nSize` is zero

Reply:
fixed